tools/alpine: Make sure we use content trust for linuxkit/alpine

rn opened the pull request:
We should always pull and push linuxkit/alpine with content trust
irrespective of architecture, unless explicitly disabled.

Currently, we have to dance around various other issues on arm64,
which are now documented in the Makefile.

Signed-off-by: Rolf Neugebauer <rolf.neugebauer@docker.com>

![tiger](https://user-images.githubusercontent.com/3338098/28581442-1e36552e-715a-11e7-98d5-b910df74fe27.jpg)

justincormack commented on tools/alpine/Makefile:
```diff
@@ -36,9 +53,9 @@ versions.$(ARCH): Makefile hash iid
 
 push: hash iid versions.$(ARCH)
-	DOCKER_CONTENT_TRUST=$(DOCKER_CONTENT_VAR) docker pull $(ORG)/$(IMAGE):$(shell cat hash) || \
+	docker pull $(ORG)/$(IMAGE):$(shell cat hash) || \
```
this is supposed to be a trusted pull if you are going to push with trust, as if there is an unsigned copy it will ignore it and replace it with your signed one.

justincormack replied:
Oh I see you are always setting trust, ok.

rn replied:
yes, but we have `export DOCKER_CONTENT_TRUST=1` set globally

justincormack commented on tools/alpine/versions.x86_64:
```diff
@@ -1,239 +0,0 @@
-# linuxkit/alpine:34af9cb1990debd17fae6d4198c62ce3910d9908
```
why are you removing this file?

rn replied:
Ah, missed that. Shouldn't remove it, but add an updated one once the overall Makefile changes are ok and I have updated the PR

justincormack commented:
Looks correct to me...

rn commented:
@arm64b note also that you need a newer version of the docker now (17.06) since the alpine base now uses `--iidfile`. I've build a master version of docker on ubuntu. My steps are here: https://gist.github.com/rn/08d13b7ed30ab5fbde9a6dcaa24831ce

rn commented:
Update the PR, now with signed ARM alpine base (weirdly, x86 did not change...I thought we'd incorporate the hash of the Makefile...). Removed the RFC bits and added animal picture
